### PR TITLE
WP-4462 add close_sinks and cancel_subscriptions lints to 3.x

### PR DESCRIPTION
**Pull request by toddbeckman-wf**
# Problem

We need to add lints to 3.x

# Solution

Add lints to 3.x and address the lints.

# Testing

- [ ] CI passes

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Raven
<details>
<summary>Number of Findings: 0</summary>

</details>

File: analysis_options.yaml [hunk 1]
```diff
@@ -18,8 +18,8 @@ linter:
     - valid_regexps
 
     # TODO: enable these when addressing memory audit/disposables
```
Comment (trentgrover-wf) [inline analysis_options.yaml]: could kill this TODO

**Comment by evanweible-wf** [comment]
+10
- [x] CI passes

**Comment by evanweible-wf** [comment]
QA +1
- [x] Testing instruction
- [x] Dev +1's
- [x] Dev/QA +10 with detail of what was tested
- [x] Unit tests created/updated
- [x] All unit tests pass

@Workiva/release-management-pp

**Comment by rmconsole-wf** [comment]
+1 from RM

**Pull request merged**